### PR TITLE
retry on  duplicate connection

### PR DESCRIPTION
**Pull request by VikrantKS**
# Issue Link
https://github.com/LambdaTest/test-at-scale/issues/118

# Description

Add retry mechanism if there is open connection error 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally with lambdatest server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/synapse/synapse.go [hunk 1]
```diff
@@ -317,26 +330,27 @@ func (s *synapse) logout() {
...
-	runnerOpts core.RunnerOptions,
+	runnerOpts *core.RunnerOptions,
 ) {
 	specs := GetResources(runnerOpts.Tier)
 	resourceStats := core.ResourceStats{
 		Status: status,
 		CPU:    specs.CPU,
 		RAM:    specs.RAM,
 	}
+	s.logger.Debugf("sending resource update to lambdatest %+v", resourceStats)
```
Comment (palashmarantas) [inline pkg/synapse/synapse.go:341]: can we add build ID here to track these logs ?

File: pkg/synapse/synapse.go [hunk 2]
```diff
@@ -172,26 +173,36 @@ func (s *synapse) messageReader(normalCloser chan struct{}, conn *websocket.Conn
...
-		_, msg, err := conn.ReadMessage()
-		if err != nil {
-			if websocket.IsCloseError(err, websocket.CloseNormalClosure) {
-				s.logger.Debugf("Normal closure occurred...........")
-				normalCloser <- struct{}{}
-				return
-			}
-			s.logger.Errorf("disconnecting from lambdatest server. error in reading message %v", err)
+		select {
+		case <-duplicateConnectionChan:
```
Comment (palashmarantas) [inline pkg/synapse/synapse.go:179]: how are we identifying duplicate connection?

Comment (VikrantKS) [reply pkg/synapse/synapse.go:179]: We are asserting over errMsg received from lambdatest server

File: pkg/synapse/synapse.go (hunk 1 above)
Comment (VikrantKS) [reply pkg/synapse/synapse.go]: added `jobID` and `buildID`

**Review by palashmarantas — APPROVED** [review]

**Pull request merged**